crypto/property/property.c

```diff
@@ -657,10 +657,13 @@ static void impl_cache_flush_one_alg(ossl_uintmax_t idx, ALGORITHM *alg,
                                      void *v)
 {
     IMPL_CACHE_FLUSH *state = (IMPL_CACHE_FLUSH *)v;
+    unsigned long orig_down_load = lh_QUERY_get_down_load(alg->cache);
 
     state->cache = alg->cache;
+    lh_QUERY_set_down_load(alg->cache, 0);
     lh_QUERY_doall_IMPL_CACHE_FLUSH(state->cache, &impl_cache_flush_cache,
                                     state);
+    lh_QUERY_set_down_load(alg->cache, orig_down_load);
```

Comment:
IIUC, this effectively means that we will never contract the size of the hashtable. It will only ever grow. Is this a problem?

Reply:
We could add a function to contract the hash table and call it after the doall iterator finishes. Not sure it is necessary as we would presumably need to grow it again anyway later. So I think this is good as is.

Reply:
alternatively we could just modify OPENSSL_LH_set_down_load such that it preforms the same check that OPENSSL_LH_delete does:
```
if ((lh->num_nodes > MIN_NODES) &&
        (lh->down_load >= (lh->num_items * LH_LOAD_MULT / lh->num_nodes)))
        contract(lh);

```
And then when we restored the down_load value after the iterator, we would get a "maybe contract" effect.  Though, I agree with @t8m in that I don't think we need to worry about unbound growth here

Reply:
Not contracting isn't a problem.  The hash table gets flushed once it exceeds the maximum size, so it cannot grow indefinitely.  It's likely better to not contract since it's only going to grow back to the maximum.  I.e. if it has to contract, it will have to expand again later.